servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/H1ProtocolConfigBuilder.java

```diff
@@ -130,14 +131,39 @@ public H1ProtocolConfigBuilder trailersEncodedSizeEstimate(final int trailersEnc
         return this;
     }
 
+    /**
+     * Defines if an HTTP/1.1 response with <a href="https://tools.ietf.org/html/rfc7230#section-6.1">
+     * Connection: close</a> and <a href="https://tools.ietf.org/html/rfc7230#section-3.3.1">
+     * Transfer-Encoding: chunked</a> headers that does not start reading the
+     * <a href="https://tools.ietf.org/html/rfc7230#section-4.1">chunked-body</a> before server closes the connection
+     * should be considered as a legit response.
+     * <p>
+     * While this use-case is not supported by <a href="https://tools.ietf.org/html/rfc7230#section-3.3.3">RFC 7230</a>,
+     * some older server implementations may use connection closure as an indicator of message completion even if
+     * {@code Transfer-Encoding: chunked} header is present:
+     * <pre>{@code
+     *     HTTP/1.1 200 OK
+     *     Content-Type: text/plain
+     *     Transfer-Encoding: chunked
+     *     Connection: close
+     * }</pre>
+     *
+     * @return {@code this}
+     */
+    public H1ProtocolConfigBuilder allowChunkedResponseWithoutBody() {
```

Comment:
It will be good if we can create a bag of spec relaxation options (expecting for more options in the future), which has the following benefits:

- Single place to enable/disable all spec relaxations (better documentation around potential dangers)
- More emphasis that these options are potentially dangerous.
- Better visibility for users when looking for unexpected behavior, eg: check if I use one method as opposed to n.

There are a few ways to approach this:

1. Use a hierarchical DSL like `clientBuilder.secure()`
2. Use an independent class like `H1SpecExceptions` which sets the flags and then provide a method here as `specExceptions(H1SpecExceptions)`
3. Use a single method like `h1SpecExceptions(boolean allowPrematureClosureWithNoPayload)` and add overloads later.

Option (2) seems better to me.